core/src/main/java/org/apache/calcite/rel/rules/JoinDeriveIsNotNullFilterRule.java

```diff
@@ -62,9 +62,9 @@ public JoinDeriveIsNotNullFilterRule(Config config) {
     final RelMetadataQuery mq = call.getMetadataQuery();
 
     final ImmutableBitSet.Builder notNullableKeys = ImmutableBitSet.builder();
-    RelOptUtil.conjunctions(join.getCondition()).forEach(node -> {
-      if (Strong.isStrong(node)) {
-        notNullableKeys.addAll(RelOptUtil.InputFinder.bits(node));
+    RelOptUtil.InputFinder.bits(join.getCondition()).forEach(bit -> {
+      if (Strong.of(ImmutableBitSet.of(bit)).isNotTrue(join.getCondition())) {
```

Comment:
This can also become a bit shorter I guess by using the static util: `Strong.isNotTrue(join.getCondition, ImmutableBitSet.of(bit))`

Reply:
Agreed.